nvme.c

```diff
@@ -659,13 +659,10 @@ static int get_effects_log(int argc, char **argv, struct command *cmd, struct pl
 	list_head_init(&log_pages);
 
 	if (cfg.csi < 0) {
-		nvme_root_t nvme_root;
 		uint64_t cap;
 		int nvme_command_set_supported;
 		int other_command_sets_supported;
-		nvme_root = nvme_scan(NULL);
-		bar = mmap_registers(nvme_root, devicename);
-		nvme_free_tree(nvme_root);
+		bar = mmap_registers(devicename);
```

Comment:
I thought, mapping registers are same (no need to get after issuing many cmds)
that's why included, do you want to remain this?
absolutely is not matter to remain or not, since it's already issuing command. for it self

Reply:
Sure, but I think this is going in the wrong direction and the commit message is speaking about `show-regs`.

Reply:
I split the commit (fabric option, and mmap_registers).
And inside of mmap_reigsters use nvme_scan_ctrls to get specified ctrl dev name.

Do you want me to move mmap_registers into libnvme? (+ print stack trace when this code path blocks)

Reply:
Your latest version looks way better. The thing I don't like is newly introduces `--fabrics` argument. 

I wonder if we could just the transport here:

```diff
+       num_ctrls = nvme_scan_ctrls(&ctrls);
+       for (i = 0; i < num_ctrls; i++) {
+               if (strncmp(dev, ctrls[i]->d_name, strlen(ctrls[i]->d_name)) == 0) {
+                       base = ctrls[i]->d_name;
+                       break;
                }
```
and read either `resource0` directly or use `nvme_get_properties()` call to figure
out the `bar`. This would keep the hack contain in one function.

Reply:
The hang is not from kernel or something.
Just device does not returning any results.
At that time, command just stop without any return.

nice Idea! I will remove that --fabrics argument.
And write down getting registers in libnvme